### DISCUSSION TITLE
How to use Make Code feature ??

### DISCUSSION THREAD

**Discussion by nitrogar** [Vector35/binaryninja-api#4705] [Q&A] [answered]
I need to disassemble a raw shellcode in the data section with no function prologue or  epilogue so make function does not work here.

I read in #1404 that the make code feature can be triggered by pressing C but it did not work.

even in the keybindings there no action associated with C  nor there is an action named make code ??

so how I can trigger this feature ?

![image](https://github.com/Vector35/binaryninja-api/assets/53944555/34dfc932-ddf0-41ec-9b57-bc849ab151f0)

**Comment by fuzyll** [comment]
My understanding, from reading back through #1404, is that those instructions are referring to [this plugin](https://github.com/Vector35/binaryninja-api/blob/dev/python/examples/make_code.py).

It's possible that, at one time, this was a feature included in the debugger (as the issue seems to suggest). But, I do not believe that is currently the case.

**Reply by nitrogar** [reply]
thanks for the fast replay. 

I just copied the code to the snippet editor and the action was created in the keybindings but still did not work.

the function make_code_helper is not triggered after pressing C

**Reply by fuzyll** [reply]
Hm, sorry to hear that - let me talk with the author of the code and see if I'm missing something.

**Comment by xusheng6** [comment, marked as answer]
Hi @nitrogar , the "make code" was once a thing but we have later removed it because it is buggy and we believe that creating a function is a superior solution. Could you please explain why that would not work?
